docs/services/postgresql/split-partitioned-database.md

```diff
@@ -0,0 +1,275 @@
+, *pg2* and *pg3*. *pg1* has one database and *pg2* and *pg3* has none:
+
+*pg1* databases:
+
+* *commcarehq_p1* (with django alias *p1*)
+
+*pg2* and *pg3* is a newly deployed server in the *[postgresql]* group and we want to
+create a new  *commcarehq_p2* on *pg2*  and *commcarehq_p3* on *pg3* with half the data from *commcarehq_p1* on each.
+
+*pg1* is currently the only database containing sharded data.
+Half of the data should be moved to a new *pg2* and *pg3* servers
+
+Current database configuration:
+
+```python
+PARTITION_DATABASE_CONFIG = {
+    'shards': {
+        'p1': [0, 3],
+    },
+    'groups': {
+        'main': ['default'],
+        'proxy': ['proxy'],
+        'form_processing': ['p1'],
+    },
+}
+
+DATABASES = {
+    'proxy': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_proxy',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+    'p1': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_p1',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+}
+```
+
+At the end of this process shards 0 & 1 should be on *pg2* and shards 2 & 3 will be on *pg3*.
+
+## Important Notes
+
+By default pglogical does not replicate any DDL commands.
+This means that any CommCareHQ migrations may not be applied to the target databases while logical replication is active.
+It is recommended to not deploy any changes during the time when splitting the database.
+More technical information can be found at https://github.com/2ndQuadrant/pglogical/blob/REL2_x_STABLE/docs/README.md#ddl
+
+## Process Overview
+
+1. Ensure that *pg1* is set to use logical replication
+2. Setup *pg2* and *pg3* as a new database
+3. Setup logical replication from pg1 to pg2 and pg3
+4. Promote *pg2* and *pg3* to a master node
+5. Update the applications configuration to go to *pg2* and *pg3*
+
+## Process detail
+
+### 1. Setup logical replication on *pg1*
+
+In your configuration set the following and deploy the configuration to *pg1*:
+
+```
+[pg1]
+...
+postgresql_wal_level = logical
+postgresql_max_worker_processes = 8
+postgresql_shared_preload_libraries = ["pglogical"]
+```
+
+Also ensure that your replication user has superuser privileges on all databases, in vault.yml:
+
+```yaml
+POSTGRES_USERS:
+  replication:
+    username: 'foo'
+    password: 'bar'
+    role_attr_flags: 'LOGIN,REPLICATION,SUPERUSER'
+```
+
+In postgresql.yml:
+
+```yaml
+postgresql_hba_entries:
+  - contype: host
+    users: foo
+    netmask: 'pg2 ip address'
+  - contype: host
+    databases: replication
+    users: foo
+    netmask: 'pg2 ip address'
```

Comment:
do you also need entries for `pg3`?

Reply:
9bf7e47665784780938844c3348e31536aafe6c5